### PR TITLE
Implement rectangular clouds with nomcica

### PR DESCRIPTION
**Pull request by SallyDa**

### CODE REVIEW

File: konrad/radiation/rrtmg.py [hunk 1]
```diff
@@ -262,6 +268,44 @@ def radiative_fluxes(self, atmosphere, surface, cloud):
 
         return lw_fluxes, sw_fluxes
 
+    def calc_cloudy_nomcica_radiation(self, atmosphere, surface, cloud):
+
+        cloud_fraction = deepcopy(cloud.cloud_area_fraction_in_atmosphere_layer)
+
+        if self._state_sw is None:  # first time only
+            cf_cloudy = cloud_fraction[cloud_fraction != 0]
+            if cf_cloudy.shape[0] != 0:
```
Comment (lkluft) [inline konrad/radiation/rrtmg.py]: Is this to ensure ``cf_cloudy.size > 0``?

File: konrad/radiation/rrtmg.py [hunk 2]
```diff
@@ -262,6 +268,44 @@ def radiative_fluxes(self, atmosphere, surface, cloud):
...
+                if not np.all(cf_cloudy == cf_cloudy[0]):
+                    logging.warning(
+                        'The konrad implementation of nomcica with partial '
+                        'cloud fraction is only suitable for a single '
+                        'rectangular cloud. Consider using mcica instead.')
```
Comment (lkluft) [inline konrad/radiation/rrtmg.py]: Is it really necessary that the cloud is rectangular or is it sufficient if it is a single cloud?

File: konrad/radiation/rrtmg.py [hunk 3]
```diff
@@ -262,6 +268,44 @@ def radiative_fluxes(self, atmosphere, surface, cloud):
...
+        lw_fluxes, sw_fluxes = lw_overcast[1], sw_overcast[1]
+
+        # Combine overcast and clear sky fluxes and heating rates to get the
+        # all sky fluxes and heating rates.
+        for fluxes in lw_fluxes, sw_fluxes:
+            for key in fluxes.keys():
+                if 'clear_sky' not in key:
+                    clear_part = fluxes[key + '_assuming_clear_sky'][:]
+                    fluxes[key][:] *= cf_max  # weighted by cloud area fraction
+                    fluxes[key][:] += (1 - cf_max) * clear_part
```
Comment (lkluft) [inline konrad/radiation/rrtmg.py]: Does this give you the net flux? If yes, does the loop overwrite both the net and the cloudy fluxes with the net fluxes?

File: konrad/radiation/rrtmg.py [hunk 4]
```diff
@@ -63,18 +65,22 @@ def init_radiative_state(self, atmosphere, surface):
 
         climt.set_constants_from_dict({"stellar_irradiance": {
                 "value": self.solar_constant, "units": 'W m^-2'}})
+        if self._mcica:
```
Comment (lkluft) [inline konrad/radiation/rrtmg.py]: I know that this is not introduced in this PR, but it is good practice to start boolean variables with a prefix like ``is_`` (e.g. ``self._is_mcica``). This helps to determine the type of the variable at first glance (at least for me).

File: konrad/radiation/rrtmg.py (hunk 2 above)
Comment (SallyDa) [reply konrad/radiation/rrtmg.py]: Actually the important point is that wherever there is cloud, the cloud fraction is the same. So, there could be two rectangular clouds with the same cloud fraction with some clear sky in between, but with the clouds on top of each other (overlap = maximum) and that would also work. The method only uses a single non-zero cloud fraction, so if the cloud fraction profile input is something else, then the cloud shape is ignored and only a single cloud fraction is used for the cloudy layers.

File: konrad/radiation/rrtmg.py (hunk 3 above)
Comment (SallyDa) [reply konrad/radiation/rrtmg.py]: Yes, this gives the net flux. The fluxes (and heating rates) stored are just for the clear sky and the net. In this method, we first calculate overcast fluxes, which are stored in the net flux variables and then over-right them with the actual net fluxes.

**Comment by SallyDa** [comment]
@lkluft Can I merge this now?

**Pull request merged**